exec/cnex/lib/posix.c

```diff
@@ -0,0 +1,781 @@
+
+void posix_seteuid(struct tagTExecutor *exec)
+{
+    Number euid = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(seteuid(number_to_sint32(euid))));
+}
+
+void posix_setgid(struct tagTExecutor *exec)
+{
+    Number gid = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(setgid(number_to_sint32(gid))));
+}
+
+void posix_setpgid(struct tagTExecutor *exec)
+{
+    Number pgid = top(exec->stack)->number; pop(exec->stack);
+    Number pid = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(setpgid(number_to_sint32(pid), number_to_sint32(pgid))));
+}
+
+void posix_setpriority(struct tagTExecutor *exec)
+{
+    Number prio = top(exec->stack)->number; pop(exec->stack);
+    Number who = top(exec->stack)->number; pop(exec->stack);
+    Number which = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(setpriority(number_to_sint32(which), number_to_sint32(who), number_to_sint32(prio))));
+}
+
+void posix_setregid(struct tagTExecutor *exec)
+{
+    Number egid = top(exec->stack)->number; pop(exec->stack);
+    Number rgid = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(setregid(number_to_sint32(rgid), number_to_sint32(egid))));
+}
+
+void posix_setreuid(struct tagTExecutor *exec)
+{
+    Number euid = top(exec->stack)->number; pop(exec->stack);
+    Number ruid = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(setreuid(number_to_sint32(ruid), number_to_sint32(euid))));
+}
+
+void posix_setsid(struct tagTExecutor *exec)
+{
+    push(exec->stack, wrap(setsid()));
+}
+
+void posix_setuid(struct tagTExecutor *exec)
+{
+    Number uid = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(setuid(number_to_sint32(uid))));
+}
+
+void posix_sleep(struct tagTExecutor *exec)
+{
+    Number seconds = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(sleep(number_to_sint32(seconds))));
+}
+
+void posix_strerror(struct tagTExecutor *exec)
+{
+    Number errnum = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, cell_fromCString(strerror(number_to_sint32(errnum))));
+}
+
+void posix_symlink(struct tagTExecutor *exec)
+{
+    char *path2 = string_asCString(top(exec->stack)->string); pop(exec->stack);
+    char *path1 = string_asCString(top(exec->stack)->string); pop(exec->stack);
+
+    push(exec->stack, wrap(symlink(path1, path2)));
+    free(path1);
+    free(path2);
+}
+
+#if HAVE_AT_FUNCTIONS
+void posix_symlinkat(struct tagTExecutor *exec)
+{
+    char *name2 = string_asCString(top(exec->stack)->string); pop(exec->stack);
+    Number fd = top(exec->stack)->number; pop(exec->stack);
+    char *name1 = string_asCString(top(exec->stack)->string); pop(exec->stack);
+
+    push(exec->stack, wrap(symlinkat(name1, number_to_sint32(fd), name2)));
+    free(name1);
+    free(name2);
+}
+#endif
+
+void posix_sync(struct tagTExecutor *exec)
+{
+    assert(exec);
```

Comment:
What is the purpose of this assert?

Reply:
That was basically because the unused exec parameter.  So, I figured an assert to assert that the value isn't NULL would have been ok here.  I can change it to simply be:
`exec=exec; // unused variable`

Reply:
A dummy assignment like that would be better, because that would actually work. An assert won't (because when you turn off asserts, the code goes away and the warning would come back).